base/math.jl

```diff
@@ -629,9 +629,17 @@ julia> hypot(3, 4im)
 hypot(x::Number, y::Number) = hypot(promote(x, y)...)
 hypot(x::Complex, y::Complex) = hypot(abs(x), abs(y))
 hypot(x::T, y::T) where {T<:Real} = hypot(float(x), float(y))
-hypot(x::T, y::T) where {T<:Number} = (z = y/x; abs(x) * sqrt(one(z) + z*z))
+function hypot(x::T, y::T) where {T<:Number}
+    # Return Inf if either or both inputs is Inf (Compliance with IEEE754)
+    if isinf(x) || isinf(y)
+        return T(Inf)
+    end
+
+    !iszero(x) ? (z = y/x; abs(x) * sqrt(oneunit(z) + z*z)) : abs(y)
```

Comment:
I would expect `oneunit(z)` and `z*z` to have incompatible units in general?

Reply:
Probably would be `oneunit(z)`?

Reply:
Should I use `oneunit(z)*oneunit(z)` in case `^` is not defined (I suppose that's why it's `z*z` instead of `z^2`)?

Reply:
I don't understand why did the tests pass with this...

Reply:
Since `z` is `y/x` it is usually unitless (the units should cancel). Are there unusual cases where that doesn't hold? If not then I think the original `one(z)` is ok.

Reply:
The unusual case is in the tests. While `y/z` is unitless, it is represented as a `Furlong{0,Float64}` and `1+z*z` gives
```
ERROR: promotion of types Int64 and Furlong{0,Float64} failed to change any arguments
```

Should I define
```
Base.promote_rule(::Type{Furlong{0,T}}, ::Type{S}) where {T, S <: Number} = Furlong{0,promote_type(S,T)}
```
or should I use `oneunit(z)*oneunit(z)`?

Reply:
I see, it uses furlongs^0 for that. Unlike Unitful.jl, but I suppose it's valid. How about
```
z = y/x; z2 = z*z; abs(x) * sqrt(oneunit(z2) + z2)
```